### PR TITLE
[20.09] nixUnstable: 2.4pre20201205_a5d85d0 -> 2.4pre20210308_1c0e3e4, hydra-unstable: 2020-10-20 -> 2021-03-10

### PR DESCRIPTION
**Pull request by Ma27**
###### Motivation for this change

Backport of #114786.
Also had to backport most recent lowdown bumps to get Nix compiling.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/hydra/common.nix [hunk 1]
```diff
@@ -84,7 +84,7 @@ in stdenv.mkDerivation rec {
       perlDeps perl nix
       postgresql # for running the tests
       nlohmann_json
-      boost
+      boost mdbook
```
Comment (jonringer) [inline pkgs/development/tools/misc/hydra/common.nix:87]: mdbook should be in nativeBuildInputs, as it's used to build documentation.

Comment (Ma27) [reply pkgs/development/tools/misc/hydra/common.nix:87]: cc @shlevy, I only applied the patches onto the release branch.

**Review by sternenseemann — COMMENTED** [review]
lowdown-related changes look good to me, the breaking changes in 0.8.0 are as follows:

* Library API changes for functions that may fail due to allocation errors. This is what `nixUnstable` actually needs and fine since no other package currently depends on lowdown as a library
* The `--nroff-hardwrap` option got removed. This is most likely a breakage fine to backport since it is reasonably obscure.

**Comment by roberth** [comment]
> no other package currently depends on lowdown as a library

Not in Nixpkgs perhaps. For a simple package like lowdown, isn't it just as easy to duplicate it, so we have a `lowdown_0_8` package next to the old `lowdown`?

File: pkgs/development/tools/misc/hydra/common.nix (hunk 1 above)
Comment (shlevy) [reply pkgs/development/tools/misc/hydra/common.nix:87]: @jonringer :facepalm: sorry, sloppy on my part

Comment (shlevy) [reply pkgs/development/tools/misc/hydra/common.nix:87]: Fixing now, just testing the build

**Comment by shlevy** [comment]
Not a big deal especially on a maintenance branch, but would be nice to have the lowdown fixes in first so we don't have a known-broken commit in the history.

File: pkgs/development/tools/misc/hydra/common.nix (hunk 1 above)
Comment (shlevy) [reply pkgs/development/tools/misc/hydra/common.nix:87]: https://github.com/NixOS/nixpkgs/commit/d47ac892331be148fd4d078c37651b4e1693414e

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 115947` at 82f018b7 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>disnix</li>
    <li>disnixos</li>
    <li>hydra-unstable</li>
    <li>nixFlakes</li>
    <li>nixUnstable</li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li><a href="https://gist.github.com/1451d5367a12345810f279a6372ff2fd">hydra-migration</a></li></li>
    <li><a href="https://gist.github.com/0ceaeda3464e267c8691b786596af5a9">lowdown</a></li></li>
  </ul>
</details>

**Comment by sternenseemann** [comment]
> Not in Nixpkgs perhaps. For a simple package like lowdown, isn't it just as easy to duplicate it, so we have a `lowdown_0_8` package next to the old `lowdown`?

FWIW the lowdown library is pretty seldomly used where the lowdown utility and nix are the only known downstream users to the library's author afaik. But agreed, a `lowdown_0_8` attribute would be the cleaner solution.

File: pkgs/development/tools/misc/hydra/common.nix (hunk 1 above)
Comment (Ma27) [reply pkgs/development/tools/misc/hydra/common.nix]: Applied here as well.

**Comment by Ma27** [comment]
> FWIW the lowdown library is pretty seldomly used where the lowdown utility and nix are the only known downstream users to the library's author afaik. But agreed, a lowdown_0_8 attribute would be the cleaner solution.

Done.

> Not a big deal especially on a maintenance branch, but would be nice to have the lowdown fixes in first so we don't have a known-broken commit in the history.

Done.

**Review by roberth — APPROVED** [review]

**Review by shlevy — APPROVED** [review]

**Pull request merged**